pyspecdata/DCCT_function.py

```diff
@@ -343,84 +252,184 @@ def place_labels(
                 label_placed[this_label_num] = 1
 
+    # }}}
+    # {{{ Generate alias labels - goes to scientific fn
+    gen_labels(my_data)
+    # }}}
+    real_data = False
+    if cmap is not None:
+        assert all(
+            np.isclose(my_data.data.imag, 0)
+        ), "In order to use a color map, you must pass real data"
+        if type(cmap) is str:
+            cmap = plt.get_cmap(cmap)
+            my_data.data = my_data.data.real
+            real_data = True
+    my_data.human_units()
+    a_shape = ndshape(this_nddata)
+    num_dims = len(a_shape.dimlabels[:-2])
+    divisions = []
+
+    # {{{ Determine number of axes objects based on shape of phase cycling
+    #     dimensions
+    # the procedure is to move from the innermost dimension to the outermost
+    # dimension, zooming out and replicating the division lines with each step.
+    for j, thisdim in enumerate(a_shape.dimlabels[::-1][2:]):
+        # I'm going to zoom out, and the division lines I add will be twice the
+        # size of the largest I have added to date, so divide all previous
+        # sizes by 2
+        old = [j / 2.0 for j in divisions]
+        # replicate the previous set of division lines, and adding a bolder
+        # division (thickness = 1 vs. the most recent which is now thickness =
+        # 1/2) between each set of replicas
+        divisions = (old + [1]) * (a_shape[thisdim] - 1) + old
+        logging.debug(strm("for", thisdim, "I get", divisions))
+    # what I have above represents the relative widths of the divisions
+    # correctly, but I want to make sure that they all add up to "gap", which
+    # is the size that I've actually allocated overall for my divisions
+    divisions = [j * gap / sum(divisions) for j in divisions]
+    # }}}
+    # {{{ Determine the bboxes for all the Axes objects that we are generating
+    # Height of axes object including gap above for space between axes objects
+    axes_height = (bbox[3] - gap) / np.prod(a_shape.shape[:-2])
+    axes_bottom = np.cumsum([axes_height + j for j in divisions])  # ndarray
+    axes_bottom = r_[0, axes_bottom]
+    # }}}
+    ax_list = []
+    # Note: Length (in fig coords) between the left most side of the figure to
+    # the left most side of the axes objects as the sum of bbox[0] and
+    # allow_for_labels which defines the length between the left side of the
+    # axes object and the left side of the decorations
+    allow_for_labels, _ = fig.transFigure.inverted().transform(
+        (horiz_label_spacer * num_dims, 0)
+    )
+    axes_bottom += bbox[1]
+    axes_width = bbox[2] - allow_for_labels
+    for j, b in enumerate(axes_bottom):
+        ax_list.append(
+            plt.axes([
+                allow_for_labels + bbox[0],
+                b,
+                axes_width,
+                axes_height,
+            ])
+        )  # lbwh
+    for ax in ax_list[1:]:
+        ax.sharex(ax_list[1])
+        ax.sharey(ax_list[1])
```

Comment:
to do said sharex and sharey first. These are not arguments for the axes in the axes list as we are making them one by one so when we say share x it doesn't understand this was AGs solution